frontend/src/components/special/Podium.tsx

```diff
@@ -0,0 +1,179 @@
+import React, { useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { Player, Submission } from '../../api/Game';
+import { Text, MediumText } from '../core/Text';
+import Language, { displayNameFromLanguage } from '../../api/Language';
+
+type PodiumProps = {
+  place: number,
+  player: Player | undefined,
+  gameStartTime: string,
+  loading: boolean,
+};
+
+type MedalProps = {
+  color: string,
+};
+
+type HeightProps = {
+  height: number,
+};
+
+const Content = styled.div`
+  display: flex;
+  flex-direction: column;
+  justify-content: flex-end;
+`;
+
+const PodiumContainer = styled.div<HeightProps>`
+  position: relative;
+  display: inline-block;
+  box-shadow: 0 1px 4px rgba(0, 0, 0, 0.12);
+  width: 180px;
+  height: ${({ height }) => height}px;
+  padding: 10px;
+  margin: 5px 10px;
+  background: ${({ theme }) => theme.colors.white};
+  border-radius: 8px;
+`;
+
+const WinnerText = styled(MediumText)`
+  font-weight: bold;
+  margin: 0;
+`;
+
+const ScoreText = styled(MediumText)`
+  font-weight: normal;
+  margin: 0;
+`;
+
+const SmallerText = styled(Text)`
+  margin: 5px 0;
+`;
+
+const BottomContent = styled.div`
+  position: absolute;
+  bottom: 5px;
+  left: 0;
+  right: 0;
+  margin: 0 auto;
+`;
+
+const Medal = styled.div<MedalProps>`
+  background: ${({ theme, color }) => theme.colors.gradients[color]};
+  border-radius: 50%;
+  margin: 15px auto;
+  
+  height: 50px;
+  width: 50px;  
+  line-height: 50px;
+`;
+
+function Podium(props: PodiumProps) {
+  const {
+    place, player, gameStartTime, loading,
+  } = props;
+
+  const [bestSubmission, setBestSubmission] = useState<Submission | null>(null);
+
+  useEffect(() => {
+    if (player) {
+      // Find best submission
+      player.submissions.forEach((submission) => {
+        if (!bestSubmission || submission.numCorrect > bestSubmission.numCorrect) {
+          setBestSubmission(submission);
+        }
+      });
+    }
+  }, [player, setBestSubmission]);
+
+  const getMedalColor = () => {
+    switch (place) {
+      case 1:
+        return 'yellow';
+      case 2:
+        return 'silver';
+      default:
+        return 'bronze';
+    }
+  };
+
+  const getPodiumHeight = () => {
+    switch (place) {
+      case 1:
+        return 280;
+      case 2:
+        return 240;
+      default:
+        return 200;
+    }
+  };
+
+  const getScoreText = () => {
+    if (!player) {
+      return <ScoreText>{loading ? 'Loading...' : 'Invite'}</ScoreText>;
```

Comment:
Yeah this was a slight annoying caveat - when the results page is first loading, this will show "loading..." and then either the player score or the "invite" text. However, this also means that when playing again it briefly shows loading again. The alternative would be to always show "invite", but then on the page initial load when players haven't been fetched yet, it will show "invite" and then potentially switch to the actual score

Reply:
Open to changing this if you have any suggestions, either in this or the next PR

Reply:
^ That's in response to the second comment - working on the first one rn

Reply:
Replaced Invite with a button that copies the invite URL:

![Button on podium to invite friends](https://user-images.githubusercontent.com/24768574/112544757-a510da80-8d74-11eb-82e0-14fe01a4f59e.png)

One thing to note is that users can't join a room that is already active - to get around this for now, I changed the ALREADY_ACTIVE error message to say "this room has already started - please wait for the host to play again" to potentially clear up confusion.

Reply:
That's a good solution, sounds good - in the future we could have a setting to allow spectators who join the game when it next starts again.

Reply:
I don't have any solutions for the "Loading..." issue, but that's a very small UI nit that we can address in the future if we feel we need to, so this is certainly ok for this PR in my mind.